core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -662,7 +667,9 @@ private void addGatewayUserFacingCertAndFqdn(GatewayConfig gatewayConfig, Cluste
 
     private List<String> getRangerFqdn(Cluster cluster, String primaryGatewayFqdn, List<String> rangerLocations) {
         if (rangerLocations.size() > 1) {
-            String gatewayGroupName = cluster.getStack().getPrimaryGatewayInstance().getInstanceGroup().getGroupName();
+            // SDX HA has multiple ranger instances in different groups, in Knox we only want to expose the ones on the gateway.
+            InstanceGroup gatewayInstanceGroup =  instanceGroupService.getPrimaryGatewayInstanceGroupByStackId(cluster.getStack().getId());
+            String gatewayGroupName = gatewayInstanceGroup.getGroupName();
```

Comment:
This will throw an NPE is `gatewayInstanceGroup` is null;

Reply:
Changed to be optional and throw an error that the instance group is not found for that stack.